src/js/components/ArticleList.js

```diff
@@ -60,6 +60,59 @@ var PanelTitle = React.createClass({
   }
 });
 
+var FilterToggler = React.createClass({
+  render: function() {
+    return (
+      <div className="btn-group" role="group">{
+        this.props.choices.map((choice, i) => {
+          return <Button key={i} label={choice.label} size="sm"
+                         type={this.props.valueCheck === choice.value ? "info" : "default"}
+                         onClick={this.props.changeHandler(choice.value)} />;
+        })
+      }</div>
+    );
+  }
+});
+
+var Filters = React.createClass({
+  statics: {
+    unreadChoices: [
+      {label: "Read", value: false},
+      {label: "Unread", value: true}
+    ],
+    statusChoices: [
+      {label: "Default", value: "0"},
+      {label: "Archived", value: "1"},
+      {label: "Deleted", value: "2"},
+    ]
+  },
+
+  filterClickHandler: function(name) {
+    return function(value) {
+      return function() {
+        ArticleActions.list({[name]: value});
```

Comment:
Yes I agree, this is handful :)

But this will end up looking like perl code  #trolldi